src/compiler/parser.ts

```diff
@@ -4010,7 +4010,20 @@ module ts {
             property.name = name;
             property.questionToken = questionToken;
             property.type = parseTypeAnnotation();
-            property.initializer = allowInAnd(parseNonParameterInitializer);
+
+            // For initializers, we always want to allow 'in' expressions. For instance properties specifically,

```

Comment:
I can remove the first sentence, or I can have the second sentence start on a new line. Would that help? The first sentence is not very interesting anyway.